app/models/generic_announcement.rb

```diff
@@ -0,0 +1,13 @@
+# Represents a generic announcement, which may be either a system-level or instance-level one.
+class GenericAnnouncement
+  include ActiveModel::Model
+
+  # @return [Array<SystemAnnouncement|Instance::Announcement>]
+  def self.valid
+    now = Time.zone.now
+    in_time_range = proc { (valid_from <= now) & (valid_to >= now) }
+    system = SystemAnnouncement.includes(:creator).where(&in_time_range)
```

Comment:
You're looking for `currently_valid`. This is an extension I defined in ActiveRecord::Base.

```ruby
system = SystemAnnouncement.includes(:creator).currently_valid
instance = ...
```

so maybe this method should be called currently_valid.